### PR TITLE
improve logging in cmd package and agent

### PR DESCRIPTION
**Pull request by Diaphteiros**
**How to categorize this PR?**

/area logging
/kind enhancement
/priority 3

Improves the logging in the `cmd` package as well as the agent.
Also fixes two minor bugs in the logging framework which cause the landscaper to panic shortly after the start.

**Release note**:

```other operator
NONE
```

### CODE REVIEW

File: cmd/landscaper-agent/app/app.go [hunk 1]
```diff
@@ -43,7 +40,7 @@ func NewLandscaperAgentCommand(ctx context.Context) *cobra.Command {
 }
 
 func (o *options) run(ctx context.Context) error {
-	o.log.Info(fmt.Sprintf("Start Landscaper Agent with version %q", version.Get().String()))
+	o.log.Info("Starting Landscaper Agent", "version", version.Get().String())
```
Comment (achimweigel) [inline cmd/landscaper-agent/app/app.go:43]: Is there a reason why you do not use lc.KeyVersion here?

File: pkg/agent/agent.go [hunk 2]
```diff
@@ -229,17 +252,22 @@ func (a *Agent) RemoveHostResources(ctx context.Context, kubeClient client.Clien
 
 // Reconcile reconciles the environment and target on the landscaper.
 func (a *Agent) Reconcile(ctx context.Context, req reconcile.Request) (reconcile.Result, error) {
+	logger, ctx := logging.MustStartReconcileFromContext(ctx, req, nil)
 	if req.Name != a.config.Name {
+		logger.Info("Not responsible for this environment", "targetEnvironment", a.config.Name)
```
Comment (achimweigel) [inline pkg/agent/agent.go:257]: Do we need a constant for targetEnvironment?

**Review by achimweigel — APPROVED** [review]
/lgtm

on two small questions

File: cmd/landscaper-agent/app/app.go (hunk 1 above)
Comment (Diaphteiros) [reply cmd/landscaper-agent/app/app.go:43]: No, maybe that was one of the first changes before the `KeyVersion` field was added. I will update it.

File: pkg/agent/agent.go (hunk 2 above)
Comment (Diaphteiros) [reply pkg/agent/agent.go:257]: Probably not, I don't think this key will appear more than once. Hm, maybe we don't even need it here, I think I add the name of the environment the agent is responsible for already to the values at a different point. Will check.

**Pull request merged**